src/EventStream.js

```diff
@@ -23,23 +29,36 @@ class EventStream extends EventEmitter {
 			const req = requestor.request({
 				hostname,
 				protocol,
-				path: `${path}?history_limit=30&access_token=${this.token}`,
+				path: `${path}?access_token=${this.token}`,
 				method: 'get',
-				port: port || (isSecure ? 443 : 80),
+				port: parseInt(port, 10) || (isSecure ? 443 : 80),
 				avoidFetch: true,
 				mode: 'prefer-streaming'
 			});
 
 			this.req = req;
-			if (this.debug) {
-				this.debug(this);
-			}
+
+			let connected = false;
+			let connectionTimeout = setTimeout(() => {
+				if (this.req) {
+					this.req.abort();
+				}
+				reject({ error: new Error('Timeout'), errorDescription: `Timeout connecting to ${this.uri}` });
+			}, this.timeout);
 
 			req.on('error', e => {
-				reject({ error: e, errorDescription: `Network error from ${this.uri}` });
+				clearTimeout(connectionTimeout);
+
+				if (connected) {
+					this.end();
```

Comment:
In the browser, it is necessary to close the connection after receiving an error.